astropy/time/formats.py

```diff
@@ -584,6 +584,16 @@ class TimeDecimalYear(TimeNumeric):
 
     name = "decimalyear"
 
+    def _check_val_type(self, val1, val2):
+        if hasattr(val1, "to") and hasattr(val1, "unit") and val1.unit is not None:
```

Comment:
```suggestion
        if hasattr(val1, "to") and getattr(val1, "unit", None) is not None:
```

Reply:
Yes, that is nicer. I should then change it for `byear` as well.

Reply:
Implemented.